drivers/ethernet/eth_stellaris.c

```diff
@@ -0,0 +1,365 @@
+	value = BIT_MACRCTL_RSTFIFO;
+	sys_write32(value, REG_MACRCTL);
+
+	/* 3. Enable transmitter */
+	value = BIT_MACTCTL_DUPLEX | BIT_MACTCTL_CRC |
+		BIT_MACTCTL_PADEN | BIT_MACTCTL_TXEN;
+	sys_write32(value, REG_MACTCTL);
+
+	/* 4. Enable Receiver */
+	value = BIT_MACRCTL_BADCRC | BIT_MACRCTL_RXEN;
+	sys_write32(value, REG_MACRCTL);
+
+	return 0;
+}
+
+static void eth_stellaris_flush(struct device *dev)
+{
+	struct eth_stellaris_runtime *dev_data = DEV_DATA(dev);
+
+	if (dev_data->tx_pos != 0) {
+		sys_write32(dev_data->tx_word, REG_MACDATA);
+		dev_data->tx_pos = 0;
+		dev_data->tx_word = 0;
+	}
+}
+
+static void eth_stellaris_send_byte(struct device *dev, u8_t byte)
+{
+	struct eth_stellaris_runtime *dev_data = DEV_DATA(dev);
+
+	dev_data->tx_word |= byte << (dev_data->tx_pos * 8);
+	dev_data->tx_pos++;
+	if (dev_data->tx_pos == 4) {
+		sys_write32(dev_data->tx_word, REG_MACDATA);
+		dev_data->tx_pos = 0;
+		dev_data->tx_word = 0;
+	}
+}
+
+static int eth_stellaris_send(struct net_if *iface, struct net_pkt *pkt)
+{
+	struct device *dev = net_if_get_device(iface);
+	struct eth_stellaris_runtime *dev_data = DEV_DATA(dev);
+	struct net_buf *frag;
+	u16_t head_len_left, i, data_len;
+	u8_t *eth_hdr;
+	struct net_eth_hdr *pkt_hdr;
+
+	if (!pkt->frags) {
+		SYS_LOG_ERR("No data to send");
+		net_pkt_unref(pkt);
+		return -ENODATA;
+	}
+
+	/* Frame transmission
+	 *  First two bytes are data_len for frame,
+	 *  Initially send the data_len
+	 */
+	data_len = net_pkt_get_len(pkt);
+	eth_stellaris_send_byte(dev, data_len & 0xff);
+	eth_stellaris_send_byte(dev, (data_len & 0xff00) >> 8);
+
+	/* Send the  header, header is 14 bytes */
+	head_len_left = net_pkt_ll_reserve(pkt);
+	eth_hdr = net_pkt_ll(pkt);
+	for (i = 0; i < head_len_left; ++i) {
+		eth_stellaris_send_byte(dev, eth_hdr[i]);
+	}
+
+	/* Send the payload */
+	for (frag = pkt->frags; frag; frag = frag->frags) {
+		for (i = 0; i < frag->len; ++i) {
+			eth_stellaris_send_byte(dev, frag->data[i]);
+		}
+	}
+	/* will transmit the partial word */
+	eth_stellaris_flush(dev);
+
+	/* Enable Txn */
+	sys_write32(BIT_MACTR_NEWTX, REG_MACTR);
+
+	/* Wait and check if txn successful or not */
+	k_sem_take(&dev_data->tx_sem, K_FOREVER);
+
+	if (dev_data->tx_err) {
+		SYS_LOG_ERR("Txn failed");
+		dev_data->tx_err = false;
+		net_pkt_unref(pkt);
+		return -EIO;
+	}
+
+	/* Update statistics counters */
+	eth_stats_update_bytes_tx(iface, net_pkt_get_len(pkt));
+	eth_stats_update_pkts_tx(iface);
+	pkt_hdr = NET_ETH_HDR(pkt);
+	if (net_eth_is_addr_multicast(&pkt_hdr->dst)) {
+		eth_stats_update_multicast_tx(iface);
+	} else if (net_eth_is_addr_broadcast(&pkt_hdr->dst)) {
+		eth_stats_update_broadcast_tx(iface);
+	}
+
+	SYS_LOG_DBG("pkt send %p len %d", pkt, net_pkt_get_len(pkt));
+	net_pkt_unref(pkt);
+
+	return 0;
+}
+
+void eth_stellaris_rx_error(struct net_if *iface)
+{
+	u32_t val;
+
+	struct device *dev = net_if_get_device(iface);
+
+	eth_stats_update_errors_rx(iface);
+	/* Clear the rx_frame buffer,
```

Comment:
Can you place empty line before the comment, it would look better that way.

Reply:
Done